src/components/notification-picker.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useCallback, useState } from 'react';
+
+import { Cluster } from './cluster';
+import {
+  Checkbox,
+  Chip,
+  FormControl,
+  FormControlLabel,
+  InputLabel,
+  MenuItem,
+  Switch,
+  TextField
+} from '@mui/material';
+import { ICreateJobModel } from '../model';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { Stack } from './stack';
+import { useTranslator } from '../hooks';
+
+interface INotificationPickerProps {
+  notificationEvents: string[];
+  id: string;
+  model: ICreateJobModel;
+  handleModelChange: (model: ICreateJobModel) => void;
```

Comment:
See if you can make the interface smaller by only passing `model.notification`, and similarly have the change handler only accept changes to that sub-object. Doing something like

```
        modelChange({
          ...model,
          notification: {
            ...model.notification,
            selectedEvents: updatedEvents
          }
        });
```

on every change is risky, since if a developer later forgets the `...model` line, then the form breaks. Keeping the props interface for a React component as small as possible is a great practice to follow.